vortex-array2/src/primitive/compute.rs

```diff
@@ -0,0 +1,25 @@
+use vortex::match_each_native_ptype;
+use vortex::scalar::Scalar;
+use vortex_error::VortexResult;
+
+use crate::compute::{ArrayCompute, ScalarAtFn};
+use crate::primitive::PrimitiveArray;
+use crate::ArrayValidity;
+
+impl ArrayCompute for &dyn PrimitiveArray {
+    fn scalar_at(&self) -> Option<&dyn ScalarAtFn> {
+        Some(self)
+    }
+}
+
+impl ScalarAtFn for &dyn PrimitiveArray {
+    fn scalar_at(&self, index: usize) -> VortexResult<Scalar> {
+        if self.is_valid(index) {
+            match_each_native_ptype!(self.ptype(), |$T| {
```

Comment:
I think if we implement accessors then we can universally implement scalar_at for all arrays with accessors by dispatching on the type and then wrapping the result in scalar. Those typed accessors would be useful in other places to have unboxed access to values.